fms/models/__init__.py

```diff
@@ -0,0 +1,50 @@
+from contextlib import nullcontext
+import os
+from pathlib import Path
+import re
+from typing import Callable, List
+import torch
+from torch import nn
+
+__models = {}
+
+
+def register_model(architecture, variant, factory):
+    variants = {}
+    if architecture in __models:
+        variants = __models[architecture]
+    variants[variant] = factory
```

Comment:
we may want to have some defensive code here for model variants that are already registered

Reply:
added